### DISCUSSION TITLE
Getting header value from a config property using @ClientHeaderParam is not working

### DISCUSSION THREAD

**Discussion by zak905** [quarkusio/quarkus#24418] [Q&A] [answered]
After trying on SO, I wanted to revert back here, before creating an issue.

I am trying to get a header value from the config into the Rest(easy) client using the `@ClientHeaderParam` annotation as described here https://quarkus.io/guides/rest-client-reactive#custom-headers-support, unfortunately it does not work out. The value is sent as-is, rather than replaced with the corresponding config property
Here is roughly what I am doing
```
@RegisterRestClient
@ClientHeaderParam(name = "Key", value = "${api-key}")
public interface MyClient {

  @POST
  @Path("/api")
  @Consumes(MediaType.APPLICATION_OCTET_STREAM)
  @Produces(MediaType.APPLICATION_JSON)
  Response call(InputStream image);
}
```

When I invoke the `call` method and check the request, I see that the `Key` header has `${api-key}` as value, and not the value I have in `application.properties` for api-key. 

According to a suggestion from a SO answer, this seems like a bug, and it normally it should be working. Is this really a bug ? or am I missing on something ?

**Comment by gsmet** [comment, marked as answer]
You are using the `quarkus-rest-client-reactive` extension, right? Not the `quarkus-rest-client` one?

**Reply by zak905** [reply]
Yes indeed that was the issue, my bad. Although, I am not using any reactive feature, I should have included the reactive client, works well now, thanks a lot!

**Reply by gsmet** [reply]
FWIW, we are making RESTEasy Reactive the default REST layer for 2.8. It works equally well for blocking or reactive workloads.

**Reply by zak905** [reply]
Good to know. Thanks
